nsuns-server/src/router.rs

```diff
@@ -42,10 +54,15 @@ where
             path_buf.push("index.html");
 
             let serve_dir = ServeDir::new(static_dir)
+                .append_index_html_on_directories(false)
                 .precompressed_gzip()
                 .precompressed_br()
                 .precompressed_deflate()
-                .not_found_service(ServeFile::new(path_buf));
+                .fallback(
+                    ServiceBuilder::new()
+                        .layer(from_fn(redirect_on_missing_auth_cookie))
```

Comment:
By layering this here, we only redirect on a missing cookie if an unknown static file is being accessed, which indicates a client-side route.